imap_processing/tests/ultra/unit/test_ultra_l1c_pset_bins.py

```diff
@@ -276,18 +276,65 @@ def test_apply_deadtime_correction(imap_ena_sim_metakernel, ancillary_files):
     # Simulate first 100 pixels are in the FOR for all spin phases
     inside_inds = 100
     spin_phase_steps[:inside_inds, :] = True
-    deadtime_ratios = np.ones(steps)
+    deadtime_ratios = xr.DataArray(np.ones(steps), dims="spin_phase_step")
 
-    pixels_below_threshold, fwhm_theta, fwhm_phi, thresholds = (
+    valid_spun_pixels, fwhm_theta, fwhm_phi, thresholds = (
         calculate_fwhm_spun_scattering(
-            spin_phase_steps, mock_theta, mock_phi, ancillary_files, 45
+            spin_phase_steps,
+            mock_theta,
+            mock_phi,
+            ancillary_files,
+            45,
+            reject_scattering=False,
         )
     )
-    boundary_sf = np.ones((pix, steps))
+    boundary_sf = xr.DataArray(np.ones((pix, steps)), dims=("pixel", "spin_phase_step"))
+    exposure_pointing_adjusted = calculate_exposure_time(
+        deadtime_ratios,
+        valid_spun_pixels,
+        boundary_sf,
+        apply_boundary_scale_factor=True,
+    )
+    # The adjusted exposure should be of shape (1,npix)
+    np.testing.assert_array_equal(exposure_pointing_adjusted.shape, (1, pix))
+    # Check that the pixels inside the FOR have adjusted exposure > 0.
+    assert np.all(exposure_pointing_adjusted[:, :inside_inds] > 0)
+    # Assert that pixels outside the FOR remain at 0.
+    assert np.all(exposure_pointing_adjusted[:, inside_inds:] == 0)
+
+
+@pytest.mark.external_kernel
+def test_apply_deadtime_correction_energy_dep(imap_ena_sim_metakernel, ancillary_files):
+    """Tests apply_deadtime_correction function when scattering rejection is on."""
+    nside = 8
+    pix = hp.nside2npix(nside)
+    steps = 500  # Reduced for testing
+    mock_theta = np.random.uniform(-60, 60, (pix, steps))
+    mock_phi = np.random.uniform(-60, 60, (pix, steps))
```

Comment:
We should be cautious about using random data in tests. Where is this used and does it even matter? We should either seed the random number generator above this, or use a linspace or similar function to sample the values in a reproducible way.